### PR TITLE
Fixes energy bolas being uniquely incapable of being blocked/caught

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Moves the ebola's snaring effect from the throw_impact proc to the ensnare proc to ensure it goes through the appropriate checks before applying the legcuffing.

## Why It's Good For The Game

Did you know that for the longest time I was convinced that bolas, in general, couldn't be caught and blocked? Turns out that since ebolas are the most commonly used bolas in the game, this impression existed only because they straight up ignored any defense you put up against bolas (except for leg loss). Oops! 

This bugs been around for a while it turns out, since it was originally a security borg item?

## Changelog
:cl:
balance:: Energy bolas no longer ignore successful blocks and catches before clapping their target with a legcuff.
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
I thought it purposely ignored it because it was an energy one, and not an actual one

**Comment by Fikou** [comment]
are you sure you tested this pr

**Comment by necromanceranne** [comment]
> I thought it purposely ignored it because it was an energy one, and not an actual one

It is so fucking old I am going to need some harder evidence than this that it is intentional to be bypassing even energy deflecting defenses.

File: code/game/objects/items/handcuffs.dm [hunk 1]
```diff
@@ -370,11 +370,11 @@
...
-	if(iscarbon(hit_atom))
-		var/obj/item/restraints/legcuffs/beartrap/B = new /obj/item/restraints/legcuffs/beartrap/energy/cyborg(get_turf(hit_atom))
-		B.Crossed(hit_atom)
-		qdel(src)
+/obj/item/restraints/legcuffs/bola/energy/ensnare(mob/living/carbon/C)
+	var/mob/living/carbon/dirty_criminal = C
+	var/obj/item/restraints/legcuffs/beartrap/justice_beartrap_that_our_bola_spawns = new /obj/item/restraints/legcuffs/beartrap/energy/cyborg(get_turf(dirty_criminal))
+	justice_beartrap_that_our_bola_spawns.Crossed(dirty_criminal)
+	qdel(src)
 	..()
```
Comment (ATH1909) [inline code/game/objects/items/handcuffs.dm:378]:
```suggestion
```

**Comment by necromanceranne** [comment]
This might be a better way to do it. Might be. Maybe. Probably.

**Comment by Mothblocks** [comment]
At this point it's been so long that I consider this a balance change, to be quite honest.

**Comment by necromanceranne** [comment]
I'll take the hit fuck it, I hate inconsistency more than anything and especially if it's throwing false info at the player.

**Comment by Mothblocks** [comment]
I updated your changelog to match, feel free to reword it if you don't like mine.

**Comment by ATH1909** [comment]
> I updated your changelog to match, feel free to reword it if you don't like mine.

who is "them"?

**Comment by necromanceranne** [comment]
That's actually my bad I'll make it clearer.

**Comment by necromanceranne** [comment]
gonna quickly draft this for later since I've run into a snare

Ha get it. Snare.

**Comment by necromanceranne** [comment]
Yeah sorry I don't quite have the time to fix this right now.

**Comment by SinfulBliss** [comment]
Kinda makes sense they can't be caught since they aren't reusable. If they become catchable there'd be no reason to use these over a craftable bola.

**Pull request closed without merging**